components/buildless-serverless/internal/controller/state/handle_git_sources.go

```diff
@@ -47,9 +47,19 @@ func sFnHandleGitSources(ctx context.Context, m *fsm.StateMachine) (fsm.StateFn,
 		return stopWithError(err)
 	}
 
+	if m.State.Function.Status.GitRepository != nil && m.State.Function.Status.GitRepository.Commit != "" &&
+		m.State.Function.Status.GitRepository.Commit != latestCommit {
```

Comment:
IMO we should update commit  and condition when status is empty.
```suggestion
	if m.State.Function.Status.GitRepository == nil || 
		m.State.Function.Status.GitRepository.Commit != latestCommit {
```